OloEngine/tests/ShaderGraph/ShaderGraphCommandTest.cpp

```
@@ -0,0 +1,215 @@
+    m_History.Execute(std::move(cmd), *m_Graph);
+    m_History.Undo(*m_Graph);
+    ASSERT_EQ(m_Graph->GetNodes().size(), 0);
+
+    m_History.Redo(*m_Graph);
+    EXPECT_EQ(m_Graph->GetNodes().size(), 1);
+    EXPECT_EQ(m_Graph->GetNodes()[0]->EditorPosition, glm::vec2(5.0f, 5.0f));
+}
+
+// ─── RemoveNodeCommand ──────────────────────────────────────
+
+TEST_F(ShaderGraphCommandTest, RemoveNodeUndoRestoresNodeAndLinks)
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)